### PR TITLE
Unstake queue

### PR DESCRIPTION
**Pull request by welldan97**
Unstake Queue functionality
- [x] Queue data fetching
- [x] Sumbitting withdrawal transaction
- [x] Tests
- [x] Docs

### CODE REVIEW

File: tests/unstakeQueue.test.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    parseEther,
+} from 'viem';
+import { Networks } from '../src/types/enums';
+import { OpusPool } from '../src';
+import { hardhat } from 'viem/chains';
+import { privateKeyToAccount } from 'viem/accounts';
+import { mine } from '@nomicfoundation/hardhat-toolbox/network-helpers';
+import { expect, test } from '@jest/globals';
+import { VaultABI } from '../src/internal/contracts/vaultAbi';
+import { resolve } from 'dns';
```
Comment (mariavarvaroi) [inline tests/unstakeQueue.test.ts:20]: Unused import

File: src/api/getUnstakeQueue.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            }
+            `,
+        variables: {
+            vault: vault.toLowerCase(),
+            receiver: pool.userAccount,
+        },
+    });
+
+    if (!queueData.data.exitRequests) {
+        // throw new Error('Queue data is missing the exitRequests field');
```
Comment (mariavarvaroi) [inline src/api/getUnstakeQueue.ts:28]: Pls throw the error

File: tests/unstakeQueue.test.ts (hunk 1 above)
Comment (welldan97) [reply tests/unstakeQueue.test.ts]: fixed

File: src/api/getUnstakeQueue.ts (hunk 2 above)
Comment (welldan97) [reply src/api/getUnstakeQueue.ts]: done

File: src/index.ts [hunk 3]
```diff
@@ -128,6 +130,45 @@ export class OpusPool {
...
+     * @param vault - A vault address
+     *
+     * @returns Array of `UnstakeQueueItem` objects corresponding to the queue
+     */
+    async getUnstakeQueueForVault(vault: Hex): Promise<Array<UnstakeQueueItem>> {
+        return getUnstakeQueue(this, vault);
+    }
+
+    /**
+     * Generates unstake transaction to withdraw from chosen Vaults unstaked queue.
```
[the remark this answers is not in the document]
Comment (welldan97) [reply src/index.ts]: fixed

File: book/docs/interfaces/UnstakeQueueItem.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+___
+
+### leftAssets
+
+• **leftAssets**: `bigint`
+
+**`Const`**
+
+Amount of assets in tokens that cannot be withdrawn
```
Comment (mksh) [inline book/docs/interfaces/UnstakeQueueItem.md:83]: I think this docstring should hint why tokens can not be withdrawn

Comment (welldan97) [reply book/docs/interfaces/UnstakeQueueItem.md:83]: Btw, do you know why they cannot be withdrawn sometimes?

I couldn't find any docs. I see this number is always equal to total shares for our test vault on holesky.

**Review by mariavarvaroi — APPROVED** [review]
LGTM! Thanks for updating the docs!

File: book/docs/interfaces/UnstakeQueueItem.md (hunk 4 above)
Comment (mksh) [reply book/docs/interfaces/UnstakeQueueItem.md:83]: Not sure, can you please ask in the Stakewise channel ?

**Review by mksh — APPROVED** [review]
LGTM, but please still contact Stakewise to clarify the docs

**Pull request merged**